doc/tutorials/understanding-error-messages.md

```diff
@@ -0,0 +1,161 @@
+   and do not call `kdbGet` before `kdbSet` since Elektra needs to know the state before it can add/modifiy/delete
+   existing keys.
+
+3. Validation `C03XXX`
+
+   Validation errors are heavily used for Elektra's `configuration specification`
+   feature and should tell users that their given input does not match a certain
+   pattern/type/expected semantic. Retry but with a conforming configuration value.
+
+## Error message format
+
+Error messages which are emitted from Elektra have the same message format for all errors
+(See [design decision](../decisions/error_message_format.md)).
+
+The design looks like:
+
+```
+Sorry, module `MODULE` issued [error|warning] `NR`:
+`ERROR_CODE_DESCRIPTION`: Validation of key "<key>" with string "<value>" failed.
+```
+
+If you want additional information about the mountpoint or the configfile which
+were affected by the error you can add the verbose argument `-v`:
+
+```
+Mountpoint: system/tests/hosts
+Configfile: /etc/hosts.13163:1575216150.11522.tmp
+```
+
+If you need more information you can pass the debug argument `-d` to the command which also shows at which line in the source code
+the error was emitted:
+
+```
+At: <path>/src/plugins/network/network.c:184
+```
+
+Error messages from tools (see [kdb.md](/doc/help/kdb.md)) may yield different messages such as when providing false command line arguments
+or mounting without incorrect permissions.
+
+Elektra's error message also includes the information about the `module` which emits the error indicated by the first line:
+`` Sorry, module `MODULE` issued [error|warning] `NR`: ``. The module indicates either indicates the core of elektra (`kdb`) or can be
+a concrete plugin. All plugins have to have a name and this is guaranteed by the compiler. Before Elektra v9.0 we also
+had even more fields to display such as `ingroup` that indicated which group emitted the error (e.g., plugins). In the effort to keep the
+message as succinct as possible we decided to [remove this field](/doc/decisions/ingroup_removal.md).
+
+## Error handling for developers
+
+If you want to contribute to Elektra or write your own specific plugin you can use the integrated error macros which
+eases the error handling. In general the [plugins README](plugins.md) is a good way to start you journey through writing your own plugin.
+All macros can be used by including `kdberrors.h` into the source file.
+For more information about the macros please read through `elektra_set_concrete_type_error` section in the [plugins.md](plugins.md) tutorial.
+Internally when Elektra emits an error it sets a [metadata](/doc/dev/metadata.md) to a Key which then is later on parsed into the error message.
+The key name of the error key is the mountpoint of the configuration. If an error occurs, 7 metadata are set which can be seen in our
+[error-handling tutorial](../dev/error-handling.md). Later on those metadata are passed into a message template which renders the message
```

Comment:
In the folder "dev" there are no tutorials but technical documents. Please distinguish these documents and clearly separate this. 
https://www.divio.com/blog/documentation/